### DISCUSSION TITLE
"Constant expression contains invalid operations" when using Laravel 10s Password validation.

### DISCUSSION THREAD

**Discussion by Simomir** [livewire/livewire#6202] [7. Bugs (archived)]
### Livewire version

v3.0-beta7

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://gitlab.com/Simomir/blog

### Steps To Reproduce

Just upgraded a small newbie project to livewire 3 and wanted to tinker with the new forms so I made one for the Register Livewire component I had already. I want to put validation rules, custom messages, attributes inside the new Rule attribute in the Register form. 
Here is the form code:
`<?php

namespace App\Livewire\Forms;

use App\Rules\FirstLetterUppercase;
use App\Rules\IsCyrilic;
use Illuminate\Validation\Rules\Password;
use Livewire\Attributes\Rule;
use Livewire\Form;

class RegisterForm extends Form
{
    #[Rule(['required', new IsCyrilic(), new FirstLetterUppercase(), 'max:100'])]
    public string $firstName = '';

    #[Rule(['required', new IsCyrilic(), new FirstLetterUppercase(), 'max:100'])]
    public string $lastName = '';

    #[Rule(['required', 'max:255', 'unique:users,email', 'email'])]
    public string $email = '';

    #[Rule(['required', Password::min(12)->mixedCase()->symbols()->numbers()->uncompromised()])]
    public string $password = '';
}`

I have no problem with custom rules I made but when it comes to the new password validation in Laravel 10 as you can see  written at the end of the code I get the following error: Constant expression contains invalid operations.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Simomir you can't pass instantiated objects (like `Password::min...`) into PHP8 attributes, only scalar types and `::class`, see https://github.com/livewire/livewire/discussions/5811#discussioncomment-6519442. So you will need to use the `rules()` method.

Hope this helps!
